cpp/src/arrow/dataset/scanner.h

```diff
@@ -104,6 +104,8 @@ struct ARROW_DS_EXPORT ScanOptions {
   ///
   /// Note: The IOContext executor will be ignored if use_threads is set to false
   io::IOContext io_context;
+  // Partition expression
+  compute::Expression partition_expression = compute::literal(true);
```

Comment:
Instead of stuffing it into the scan options, it might be better to thread this through from SkyhookFileFormat into SkyhookScanTask and then into the serialized scan request.